modules/grpc/otel/filterx/otel-field.cpp

```diff
@@ -269,7 +272,7 @@ class OtelDatetimeConverter : public ProtobufField
   FilterXObject *FilterXObjectGetter(Message *message, ProtoReflectors reflectors)
   {
     uint64_t val = reflectors.reflection->GetUInt64(*message, reflectors.fieldDescriptor);
-    UnixTime utime = unix_time_from_unix_epoch(val);
+    UnixTime utime = unix_time_from_unix_epoch(NSEC_TO_USEC(val));
```

Comment:
with this, I'd probably introduce the unix_time_from_unix_epoch_nsec() function, so we can push the NSEC_TO_USEC to be within the function body and not here.